backend/src/main/java/de/amos/apachepulsarui/service/MessageService.java

```diff
@@ -12,35 +12,44 @@
 import lombok.extern.slf4j.Slf4j;
 import org.apache.pulsar.client.admin.PulsarAdmin;
 import org.apache.pulsar.client.admin.PulsarAdminException;
-import org.apache.pulsar.client.api.Producer;
+import org.apache.pulsar.client.api.Message;
 import org.apache.pulsar.client.api.PulsarClient;
-import org.apache.pulsar.client.api.PulsarClientException;
+import org.apache.pulsar.common.api.proto.CommandSubscribe;
 import org.apache.pulsar.common.naming.TopicName;
 import org.springframework.stereotype.Service;
 
-import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
 import java.util.List;
 
 @Service
 @Slf4j
 @RequiredArgsConstructor
 public class MessageService {
-
-    public static final int MAX_NUM_MESSAGES = 20;
+    private final PulsarAdmin pulsarAdmin;
 
     private final PulsarClient pulsarClient;
-    private final PulsarAdmin pulsarAdmin;
 
-    public List<MessageDto> peekMessages(String topic, String subscription) throws PulsarApiException {
+    public List<MessageDto> getNumberOfLatestMessagesFromTopic(String topic, Integer numOfLatestMsgs) {
```

Comment:
From my point of view, this name implies that I get the amount of latest messages and not the messages itself.
```suggestion
    public List<MessageDto> getLatestMessagesOfTopic(String topic, Integer numOfLatestMsgs) {
```